clients/nodejs/exp-client/client.ts

```diff
@@ -0,0 +1,165 @@
+    tenant: string | null = null;
+    pollingFrequency: number = 10;
+    cacHostName: string | null = null;
+    delimeter: string = ",";
+
+    rustLib = ffi.Library(libPath, {
+        'expt_new_client': [int, [string, int, string]],
+        'expt_start_polling_update': [voidType, [string]],
+        'expt_get_client': ["pointer", [string]],
+        'expt_get_applicable_variant': [string, ["pointer", string, int]],
+        'expt_get_satisfied_experiments': [string, ["pointer", string, "pointer"]],
+        'expt_get_filtered_satisfied_experiments': [string, ["pointer", string, "pointer"]],
+        'expt_get_running_experiments': [string, ["pointer"]],
+        'expt_free_string': [voidType, [string]],
+        'expt_last_error_message': [string, []],
+        'expt_last_error_length': [int, []],
+        'expt_free_client': [voidType, ["pointer"]]
+    });
+
+    constructor(tenantName: string, pollingFrequency: number, hostName: string) {
+        this.tenant = tenantName;
+        this.cacHostName = hostName;
+        this.pollingFrequency = pollingFrequency;
+    }
+
+    public getExperimentationLastErrorMessage(): string | null {
+        return this.rustLib.expt_last_error_message();
+    }
+
+    public createNewExperimentaionClient(): number {
+        let respCode = this.rustLib.expt_new_client(
+            this.tenant, this.pollingFrequency, this.cacHostName
+        );
+        if (respCode == 1) {
+            let errorMessage = this.getExperimentationLastErrorMessage();
+            console.log("Some Error Occured while creating new experimentation client ", errorMessage);
+            throw Error("Client Creation Error");
+        }
+        return respCode;
+    }
+
+    public getExperimentationClient(): ref.Pointer<unknown> {
+        return this.rustLib.expt_get_client(this.tenant);
+    }
+
+    public getRunningExpriments(): string | null {
+        let clientPtr = this.getExperimentationClient();
+        return this.rustLib.expt_get_running_experiments(clientPtr);
+    }
+
+    public freeString(str: string) {
+        this.rustLib.expt_free_string(str);
+    }
+
+    public async startExperimantionPollingUpdate() {
+        if (isMainThread && worker) {
+            worker.postMessage(
+                { tenant: this.tenant
+                , event : "startExperimantionPollingUpdate"
+                , pollingFrequency: this.pollingFrequency
+                , cacHostName: this.cacHostName
+                }
+            );
+            return;
+        }
+        if (!isMainThread) {
+            this.rustLib.expt_start_polling_update(this.tenant)
+        }
+    }
+
+    public getExperimentationLastErrorLength(): number {
+        return this.rustLib.expt_last_error_length()
+    }
+
+    public freeExprementationClient() {
+        this.rustLib.expt_free_client(this.getExperimentationClient());
+    }
+
+    public getFilteredSatisfiedExperiments(context: Object, filterPrefix: string[] | null): string | null {
```

Comment:
places which take input of the form: `some_type` | `null`, should ideally take `some_type` | `undefined` type data